shipyard_blueprints/mysql/shipyard_mysql/cli/upload_file.py

```diff
@@ -60,119 +50,73 @@ def get_args():
-    This will override system defaults.
-    """
-    if args.db_connection_url:
-        os.environ["DB_CONNECTION_URL"] = args.db_connection_url
-    elif args.host and args.username and args.database:
-        os.environ["DB_CONNECTION_URL"] = (
-            f"mysql+mysqlconnector://{args.username}:{args.password}@{args.host}:{args.port}/{args.database}?{args.url_parameters}"
+def main():
+    try:
+        args = get_args()
+        match_type = args.source_file_name_match_type
+        src_file = args.source_file_name
+        src_dir = args.source_folder_name
+        src_path = shipyard.files.combine_folder_and_file_name(
+            folder_name=src_dir, file_name=src_file
         )
+        table_name = args.table_name
+        insert_method = args.insert_method
+        # client args
+        client_args = {
+            "username": args.username,
+            "pwd": args.password,
+            "host": args.host,
+            "database": args.database,
+            "port": args.port,
+            "url_params": args.url_parameters if args.url_parameters != "" else None,
+        }
+
+        mysql = MySqlClient(**client_args)
+
+        if match_type == "regex_match":
+            file_names = shipyard.files.find_all_local_file_names(src_dir)
+            matching_file_names = shipyard.files.find_all_file_matches(
+                file_names, re.compile(src_file)
+            )
+            if n_matches := len(matching_file_names) == 0:
+                logger.error("No files found matching the regex provided")
+                sys.exit(Database.EXIT_CODE_NO_FILE_MATCHES)
 
-    db_string = os.environ.get("DB_CONNECTION_URL")
-    return db_string
-
-
-def find_all_local_file_names(source_folder_name):
-    """
-    Returns a list of all files that exist in the current working directory,
-    filtered by source_folder_name if provided.
-    """
-    cwd = os.getcwd()
-    cwd_extension = os.path.normpath(f"{cwd}/{source_folder_name}/**")
-    file_names = glob.glob(cwd_extension, recursive=True)
-    return [file_name for file_name in file_names if os.path.isfile(file_name)]
-
-
-def find_all_file_matches(file_names, file_name_re):
-    """
-    Return a list of all file_names that matched the regular expression.
-    """
-    matching_file_names = []
-    for file in file_names:
-        if re.search(file_name_re, file):
-            matching_file_names.append(file)
-
-    return matching_file_names
-
-
-def combine_folder_and_file_name(folder_name, file_name):
-    """
-    Combine together the provided folder_name and file_name into one path variable.
-    """
-    combined_name = os.path.normpath(
-        f'{folder_name}{"/" if folder_name else ""}{file_name}'
-    )
+            logger.info(f"{n_matches} files found. Preparing to upload...")
 
-    return combined_name
+            for index, key_name in enumerate(matching_file_names):
```

Comment:
I could be missing something very obvious but and this comment would apply to the prs but shouldn't this be
```
for file in matching_file_names:
      mysql.upload(file=file,table_name = table_name, insert_method=insert_method)
      ...
```

Reply:
is enumerate needed?

Reply:
No [](edited-with-Axolo)